### DISCUSSION TITLE
Error setup.py:416 while first try to setup

### DISCUSSION THREAD

**Discussion by Stephan-4711** [RogerSelwyn/O365-HomeAssistant#259] [Q&A]
Hi, 

I tried to setup my ExOnline Mailbox.
I used HACS for installation and created a folder o365/ where I put the O365.yaml with the example config
and in the configuration.yaml I added:
o365:  !include_dir_merge_list o365/

developer console doesn't give an error. But on HA restart I get this log output

`Logger: homeassistant.setup
Source: setup.py:416
First occurred: 11:11:41 AM (1 occurrences)
Last logged: 11:11:41 AM

Error during setup of component o365
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 416, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/o365/__init__.py", line 48, in async_setup
    accounts = MULTI_ACCOUNT_SCHEMA(conf)[CONF_ACCOUNTS]
               ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'accounts'`

**Comment by RogerSelwyn** [comment]
This is really a question on how to use the HA configuration.yaml properly. 

What you have done is put the o365: in twice, once in configuration.yaml and once in O365.yaml. Just remove the one from O365.yaml, and move everything to the left to space.

**Comment by Stephan-4711** [comment]
Was just the spaces THX

But I switched to MS365 Calender (got also a 500error like in an other issue)
If it's working I'll add Mail too.

**Reply by RogerSelwyn** [reply]
Assuming you are using a personal account, please note the comment I have put on the prerequisites page of the docs. 

_Note - Personal accounts
Since the middle of 2024, Microsoft has mandated that any new app registrations must be created within an Entra ID directory. You will need to sign up for [Azure](https://azure.microsoft.com/en-gb/free) on a pay as you go basis, where Microsoft Entra ID is indicated as always being free._

**Reply by Stephan-4711** [reply]
No

**Comment by Stephan-4711** [comment]
What I don't understand. why there is the o365 Integration and the MS365 Calender and Email and ToDo?

**Reply by RogerSelwyn** [reply]
https://github.com/RogerSelwyn/O365-HomeAssistant/discussions/234

**Reply by Stephan-4711** [reply]
OK THanks, the calender integration works. 

The fact that the Global Admin needs a mailbox is somewhat unfortunate. Perhaps this should be pointed out during the setup process. It might also be good to see if you can find a workaround for this. Maybe switch to this issue: [https://github.com/RogerSelwyn/MS365-Calendar/issues/26#issuecomment-2410661024]
